### PR TITLE
Refactor SszPrimitive classes to enable custom SSZ primitive types

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

Enable SszUInt64 extension to custom types such as `Slot`, `Epoch` etc 
- Remove extra generic parametrization from `SszPrimitive` class to reduce hierarchy typing complexity
- Extract anonymous `UINT64_SCHEMA` to a separate `AbstractSszUInt64Schema`

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by Nashatyrev** [comment]
With the e06f200 commit I erroneously added 'this' escape from constructor, which is basically a bad practice. 
The tests failures were triggered not by the scape itself though. Initialization of json type definition from the `AbstractSszUint64Schema` made a loop in class initializers `SszPrimitiveSchemas` <=> `SszPrimitiveTypeDefinitions` and we got some nulls in place of static fields  during < clinit>

File: infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/impl/AbstractSszPrimitiveSchema.java [hunk 1]
```diff
@@ -37,10 +38,25 @@
  *
  * @param <SszDataT> Class of the basic view of this type
  */
-public abstract class AbstractSszPrimitiveSchema<
-        DataT, SszDataT extends SszPrimitive<DataT, SszDataT>>
+public abstract class AbstractSszPrimitiveSchema<DataT, SszDataT extends SszPrimitive<DataT>>
     implements SszPrimitiveSchema<DataT, SszDataT> {
 
+  protected static Bytes updateExtending(Bytes origBytes, int origOff, Bytes newBytes) {
```
Comment (rolfyone) [inline infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/impl/AbstractSszPrimitiveSchema.java]: im surprised you got away with not using `final`, but we should add...

File: infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/impl/AbstractSszUInt64Schema.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  private Supplier<DeserializableTypeDefinition<T>> typeDefinition =
+      Suppliers.memoize(this::createJsonTypeDefinition);
+
+  protected AbstractSszUInt64Schema() {
+    super(64);
+  }
+
+  @Override
+  public UInt64 createFromLeafBackingNode(LeafDataNode node, int internalIndex) {
```
Comment (rolfyone) [inline infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/impl/AbstractSszUInt64Schema.java]: does look like `final` in a few places...

File: infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/impl/AbstractSszPrimitiveSchema.java (hunk 1 above)
Comment (Nashatyrev) [reply infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/impl/AbstractSszPrimitiveSchema.java]: I hope I fixed this everywhere. 
I believe 'final' was optional when I was doing my last PR )))

**Review by tbenr — APPROVED** [review]
LGTM

**Comment by Nashatyrev** [comment]
Guys, thanks for reviewing and merging 👍

**Pull request merged**